src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -230,10 +225,37 @@ public static JsonRpc Attach(Stream sendingStream, Stream receivingStream, objec
         }
 
         /// <summary>
+        /// Adds the specified targets as possible objects to invoke when incoming messages are received.
+        /// </summary>
+        /// <param name="targets">Targets to invoke when incoming messages are received.</param>
+        /// <remarks>This method must be called before JsonRpc starts listening for messages.</remarks>
+        public void AddLocalRpcTarget(params object[] targets)
+        {
+            Requires.NotNull(targets, nameof(targets));
+
+            if (this.startedListening)
+            {
+                throw new InvalidOperationException("Cannot attach additional targets once JsonRpc has started listening for messages.");
+            }
+
+            foreach (var target in targets)
+            {
+                if (target != null)
+                {
+                    this.targetRequestMethodToClrMethodMap.Add(new Tuple<object, ReadOnlyDictionary<string, string>>(
```

Comment:
Tip: you can use `Tuple.Create(` instead of `new` here to avoid having to specify all the generic type arguments since the inferred type arguments will be equivalent.